db/src/db.rs

```diff
@@ -457,17 +462,149 @@ pub enum SearchType {
     Inexact,
 }
 
-impl DB {
-
+/// `MentatStoring` will be the trait that encapsulates the storage layer.  It is consumed by the
+/// transaction processing layer.
+///
+/// Right now, the only implementation of `MentatStoring` is the SQLite-specific SQL schema.  In the
+/// future, we might consider other SQL engines (perhaps with different fulltext indexing), or
+/// entirely different data stores, say ones shaped like key-value stores.
+pub trait MentatStoring {
```

Comment:
@rnewman set the trend with `SchemaBuilding` rather than `SchemaBuilder`.  I think I gently suggested nouns, but I've continued with the verbs for now.  Maybe we should discuss this in a meeting, since I see no guidance in the [Rust Guidelines](https://aturon.github.io/features/traits/README.html).